### PR TITLE
Migrate to the AWS Java CDK v2

### PR DESCRIPTION
**Pull request by rieckpil**
Migrating to the latest version of `cdk-constructs` and using the AWS CDK v2.

Left todos:

- [x] exclude the additional import of the CDK libraries as they're transitively pulled by the CDK

### CODE REVIEW

File: cdk/pom.xml [hunk 1]
```diff
@@ -11,11 +11,11 @@
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <cdk.version>1.121.0</cdk.version>
     <junit.version>5.7.0</junit.version>
     <passay.version>1.6.0</passay.version>
-    <cdk-constructs.version>0.0.28</cdk-constructs.version>
+    <cdk-constructs.version>2.0.0-SNAPSHOT</cdk-constructs.version> <!-- TODO: Adjust with CDK v2 compatible version -->
```
Comment (rieckpil) [inline cdk/pom.xml:16]: Waiting on the release of a AWS CDK v2 compatible release of `cdk-constructs` https://github.com/stratospheric-dev/cdk-constructs/pull/21

File: cdk/package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "canary:destroy": "cdk destroy --app \"./mvnw -e -q compile exec:java -Dexec.mainClass=dev.stratospheric.todoapp.cdk.CanaryApp\" --force --require-approval never"
   },
   "devDependencies": {
-    "aws-cdk": "1.124.0"
+    "aws-cdk": "2.5.0"
```
Comment (BjoernKW) [inline cdk/package.json:34]:
```suggestion
    "aws-cdk": "2.7.0"
```

File: chapters/chapter-10/cdk/package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "cognito:destroy": "cdk destroy --app \"./mvnw -e -q compile exec:java -Dexec.mainClass=dev.stratospheric.todoapp.cdk.CognitoApp\" --force --require-approval never"
   },
   "devDependencies": {
-    "aws-cdk": "1.124.0"
+    "aws-cdk": "2.5.0"
```
Comment (BjoernKW) [inline chapters/chapter-10/cdk/package.json:18]:
```suggestion
    "aws-cdk": "2.7.0"
```

File: chapters/chapter-11/cdk/pom.xml [hunk 4]
```diff
@@ -12,9 +12,8 @@
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <cdk.version>1.124.0</cdk.version>
     <junit.version>5.7.0</junit.version>
-    <cdk-constructs.version>0.0.28</cdk-constructs.version>
+    <cdk-constructs.version>0.1.0</cdk-constructs.version>
     <log4j2.version>2.17.0</log4j2.version>
```
Comment (BjoernKW) [inline chapters/chapter-11/cdk/pom.xml:17]:
```suggestion
    <log4j2.version>2.17.1</log4j2.version>
```

File: chapters/chapter-10/cdk/pom.xml [hunk 5]
```diff
@@ -12,10 +12,10 @@
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <cdk.version>1.124.0</cdk.version>
     <junit.version>5.7.0</junit.version>
-    <cdk-constructs.version>0.0.28</cdk-constructs.version>
+    <cdk-constructs.version>0.1.0</cdk-constructs.version>
     <log4j2.version>2.17.0</log4j2.version>
```
Comment (BjoernKW) [inline chapters/chapter-10/cdk/pom.xml:17]:
```suggestion
    <log4j2.version>2.17.1</log4j2.version>
```

File: cdk/pom.xml [hunk 6]
```diff
@@ -11,10 +11,9 @@
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <cdk.version>1.121.0</cdk.version>
     <junit.version>5.7.0</junit.version>
     <passay.version>1.6.0</passay.version>
-    <cdk-constructs.version>0.0.28</cdk-constructs.version>
+    <cdk-constructs.version>0.1.0</cdk-constructs.version>
     <log4j2.version>2.17.0</log4j2.version>
```
Comment (BjoernKW) [inline cdk/pom.xml:17]:
```suggestion
    <log4j2.version>2.17.1</log4j2.version>
```

File: chapters/chapter-12/cdk/pom.xml [hunk 7]
```diff
@@ -12,9 +12,8 @@
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <cdk.version>1.124.0</cdk.version>
     <junit.version>5.7.0</junit.version>
-    <cdk-constructs.version>0.0.28</cdk-constructs.version>
+    <cdk-constructs.version>0.1.0</cdk-constructs.version>
     <log4j2.version>2.17.0</log4j2.version>
```
Comment (BjoernKW) [inline chapters/chapter-12/cdk/pom.xml:17]:
```suggestion
    <log4j2.version>2.17.1</log4j2.version>
```

File: chapters/chapter-11/cdk/package.json [hunk 8]
```diff
@@ -15,7 +15,7 @@
     "database:destroy": "cdk destroy --app \"./mvnw -e -q compile exec:java -Dexec.mainClass=dev.stratospheric.todoapp.cdk.DatabaseApp\" --force --require-approval never"
   },
   "devDependencies": {
-    "aws-cdk": "1.124.0"
+    "aws-cdk": "2.5.0"
```
Comment (BjoernKW) [inline chapters/chapter-11/cdk/package.json:18]:
```suggestion
    "aws-cdk": "2.7.0"
```

**Review by BjoernKW — APPROVED** [review]

**Review by BjoernKW — APPROVED** [review]

**Comment by rieckpil** [comment]
AWS CDK releases too fast, thanks for the additional version bumps :D

**Review by BjoernKW — APPROVED** [review]

**Pull request merged**